google/cloud/pubsub_v1/subscriber/message.py

```diff
@@ -242,9 +243,43 @@ def ack(self) -> None:
                 byte_size=self.size,
                 time_to_ack=time_to_ack,
                 ordering_key=self.ordering_key,
+                future=None,
             )
         )
 
+    def ack_with_response(self) -> "futures.Future":
+        """Acknowledge the given message.
+
+        Acknowledging a message in Pub/Sub means that you are done
+        with it, and it will not be delivered to this subscription again.
+        You should avoid acknowledging messages until you have
+        *finished* processing them, so that in the event of a failure,
+        you receive the message again.
+
+        .. warning::
+            Acks in Pub/Sub are best effort. You should always
```

Comment:
With exactly once this isn't correct. Please add what happens when users use this method with and without exactly once.

Reply:
Fixed.